src/app/shared/dot-save-on-deactivate-service/dot-save-on-deactivate.service.spec.ts

```diff
@@ -0,0 +1,100 @@
+import { DotSaveOnDeactivateService } from './dot-save-on-deactivate.service';
+import { DotDialogService } from '../../api/services/dot-dialog/dot-dialog.service';
+import { DOTTestBed } from '../../test/dot-test-bed';
+import { LoginService, ResponseView } from 'dotcms-js/dotcms-js';
+import { LoginServiceMock } from '../../test/login-service.mock';
+import { OnSaveDeactivate } from './save-on-deactivate';
+import { Observable } from 'rxjs/Observable';
+import { DotDialog } from '../models/dot-confirmation/dot-confirmation.model';
+import { Component } from '@angular/core';
+import {mockResponseView} from '../../test/response-view.mock';
+
+@Component({
+    selector: 'dot-test',
+    template: '<h1>Test</h1>'
+})
+class MockComponent implements OnSaveDeactivate {
+    isModelChanged(): boolean {
+        return true;
+    }
+
+    onDeactivateSave(): Observable<any> {
+        return Observable.of('Saved');
+    }
+
+    onDeactivateSaveError(response: ResponseView): void {
+        throw new Error('Method not implemented.');
+    }
+
+    getSaveWarningMessages(): DotDialog {
+        return { header: 'Header', message: 'message' };
+    }
+}
+
+describe('DotSaveOnDeactivateService', () => {
+    let dotSaveOnDeactivateService: DotSaveOnDeactivateService;
+    let mockComponent: MockComponent;
+    let dotDialogService: DotDialogService;
+    beforeEach(() => {
+        const testbed = DOTTestBed.configureTestingModule({
+            declarations: [MockComponent],
+            providers: [
+                DotSaveOnDeactivateService,
+                DotDialogService,
+                {
+                    provide: LoginService,
+                    useClass: LoginServiceMock
+                }
+            ],
+            imports: []
+        });
+        dotSaveOnDeactivateService = testbed.get(DotSaveOnDeactivateService);
+        dotDialogService = testbed.get(DotDialogService);
+        mockComponent = new MockComponent();
+    });
+
+    it('should return true if there is not changes in the model', () => {
```

Comment:
Double check the description of the tests in some says Observable of true and other just true.